README.md

```diff
@@ -180,6 +180,31 @@ To run the E2E tests, you can find a script in `tools/` that will build everythi
 ./tools/run-e2e-test.sh Apple/Simulator.Tests.proj
 ```
 
+## Troubleshooting
+
+Some XHarness commands only work in some scenarios and it's good to know what to expect from the tool.
+Some Android/Apple versions also require some workarounds and those are also good to know about.
+
+### My unit tests are not running
+
+For the `test` command, XHarness expects the application to contain a `TestRunner` which is a library you can find in this repository.
+This library executes unit tests similarly how you would execute them on other platforms.
+However, the `TestRunner` from this repository contains more mechanisms that help to work around some issues (mostly in Apple platforms).
+
+The way it works is that XHarness usually sets some [environmental variables](https://github.com/dotnet/xharness/blob/main/src/Microsoft.DotNet.XHarness.iOS.Shared/Execution/EnviromentVariables.cs) for the application and the [`TestRunner` recognizes them](https://github.com/dotnet/xharness/blob/main/src/Microsoft.DotNet.XHarness.TestRunners.Common/ApplicationOptions.cs) and acts upon them.
```

Comment:
I think it's worth to mention that environmental variables are available for apple only

Reply:
Good point. Fixed